pkg/fieldpath/paved.go

```diff
@@ -220,7 +220,15 @@ func (p *Paved) GetBool(path string) (bool, error) {
 	return b, nil
 }
 
+// NOTE(muvaf): If there is no CRD, unstructured.Unstructured reads numbers as
+// float64. However, in practice, use of float64 is discouraged and when you fetch
```

Comment:
I had a little trouble parsing this. Are you saying that if you paved a `struct` with `int64` fields then unstructured is going to treat them as `float64`, but if you get an object from the api-server into `unstructured.Unstructured` then it is going to treat them as `int64`? I am guessing this is due to an issue with `runtime.DefaultUnstructuredConverter` (which we use above https://github.com/crossplane/crossplane-runtime/pull/199/files#diff-1c5d867f3ffb167cee0af55ba02e05e4R53) I worked on with some of the Velero folks, which has now been fixed upstream, but will not be included until `1.20` release.

Issue: https://github.com/kubernetes/kubernetes/issues/87675
PR: https://github.com/kubernetes/kubernetes/pull/93250

Anyway, not sure this directly impacts your change here, but might be worth leaving a note. We could avoid this by doing something like https://github.com/crossplane/crossplane/blob/b6c6ae4c6b64d29866e70d8e20f5dee3e07f97e2/pkg/parser/conversion.go#L30

Reply:
For example, if we supply `{"metadata":{"version":2}}`, it [will show that](https://github.com/crossplane/crossplane-runtime/blob/bd09d0c/pkg/fieldpath/paved_test.go#L491) the Go type of `metadata.version` is `float64`. However, if you run this code against an api-server, it will treat a numerical field as `int64`. I made a connection there about CRD in the api-server, i.e. api-server might be sending additional type information since it knows but it could be due to something else, too. In the end, I think it's probably not possible to retrieve `float64` from api-server when using `Unstructured`. @hasheddan is this what's fixed there?